### PR TITLE
[TASK] Introduce functional tests

### PR DESCRIPTION
**Pull request by linawolf**

### CODE REVIEW

**Review by froemken — CHANGES_REQUESTED** [review]

File: Tests/Functional/Domain/Repository/PostRepositoryTest.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace T3docs\blog_example\Tests\Functional\Domain\Repository;
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Tests/Functional/Domain/Repository/PostRepositoryTest.php]: Woops, good catch!

**Comment by linawolf** [comment]
Thanks a lot for your contributions! However I would like to keep  the changes suggested to the runtests.sh suggested by @sbuerk , so I will revert these changes.

File: Build/php-cs-fixer/config.php [hunk 2]
```diff
@@ -2,20 +2,58 @@
 
 declare(strict_types=1);
 
+/*
```
Comment (linawolf) [inline Build/php-cs-fixer/config.php:5]: I dont think we need theser disclaimer in configuration files or TCA. Should be enough to put them in classes?

**Comment by sbuerk** [comment]
@froemken @linawolf 

I added a dedicated pull-request introducing generic
unit and functional testing infrastructure, so this
can be kept for now and @froemken extract his other
changes, for example cgl and configuration file places
in a dedicated change/pull-request.

See [#108 [TASK] Introduce unit and functional testing infrastructure](https://github.com/TYPO3-Documentation/blog_example/pull/108)

**Comment by linawolf** [comment]
@froemken I think I moved most of your changes in separate PRs and merged them. About the remaining changes I am not so sure.
While your phpstorm.meta file looks interesting and I will try it out I think it is related to one special IDE and not at all to this project. So that would be another thing we had to test and maintain. So it does not really belong in this extension. You can keep one locally if you want.

Also I am reculant to spread the copy right information all over all PHP files. We display these files by automatic inclusion in the docs and it would blow up the docs.. I would rather just include a general LICENSE file or some such..

Can you check what else is still missing from your PR?

**Pull request closed without merging**